src/config/applicationConfig.ts

```
@@ -0,0 +1,584 @@
+        fs.chmodSync(configPath, 0o600);
+    } catch (error) {
+        if ((error as NodeJS.ErrnoException).code !== "EEXIST") {
+            throw new ApplicationConfigError(`Unable to create default config at ${configPath}`, error);
+        }
+    } finally {
+        if (temporaryPath) {
+            try {
+                fs.unlinkSync(temporaryPath);
+            } catch (error) {
+                if ((error as NodeJS.ErrnoException).code !== "ENOENT") {
+                    throw new ApplicationConfigError(`Unable to clean up temporary config at ${temporaryPath}`, error);
```

Biome (2.5.5)

[error] Unsafe usage of 'throw'.

(lint/correctness/noUnsafeFinally)